web/src/message_summary.ts

```diff
@@ -17,21 +18,36 @@ export function get_narrow_summary(channel_id: number, topic_name: string): void
         {operator: "topic", operand: topic_name},
     ]);
     const data = {narrow: message_fetch.get_narrow_for_message_fetch(filter)};
-    const channel_name = stream_data.get_stream_name_from_id(channel_id);
     const display_topic_name = util.get_final_topic_display_name(topic_name);
-    dialog_widget.launch({
-        html_heading: $t(
-            {defaultMessage: "Summary of #${channel_name} > ${topic_name}:"},
-            {channel_name, topic_name: display_topic_name},
-        ),
-        html_body: "",
-        html_submit_button: "Close",
-        close_on_submit: true,
+    const unread_topic_params = {
+        html_submit_button: $t({defaultMessage: "Mark topic as read"}),
+        html_exit_button: $t({defaultMessage: "Close"}),
+        on_click() {
+            unread_ops.mark_topic_as_read(channel_id, topic_name);
+        },
+        single_footer_button: false,
+    };
+
+    let params = {
+        html_submit_button: $t({defaultMessage: "Close"}),
         on_click() {
             // Just close the modal, there is nothing else to do.
         },
-        id: "topic-summary-modal",
         single_footer_button: true,
+    };
+    if (unread.topic_has_any_unread(channel_id, topic_name)) {
+        params = {
+            ...params,
+            ...unread_topic_params,
+        };
+    }
+    dialog_widget.launch({
+        html_heading: $t({defaultMessage: "{topic_name}"}, {topic_name: display_topic_name}),
```

Comment:
This is a weird translation tag; it'll make translators need to.

I think maybe the right answer is to have a `text_heading` alternative field that just goes into the template using `{{ html_heading }}` rather than `{{{`.

Reply:
Done.